great_expectations/rule_based_profiler/config/base.py

```diff
@@ -77,11 +64,17 @@ def remove_nulls(self, data: dict, **kwargs) -> dict:
         return cleaned_data
 
 
-@dataclass(frozen=True)
 class DomainBuilderConfig(DictDot):
-    class_name: str
-    module_name: Optional[str] = None
-    batch_request: Optional[Dict[str, Any]] = None
+    def __init__(
+        self,
+        class_name: str,
+        module_name: Optional[str] = None,
+        batch_request: Optional[Dict[str, Any]] = None,
+        **kwargs
```

Comment:
Simply ignore kwargs

Reply:
@cdkini Why not set them?  Because the user of DomainBuilder will know they are passed and may want to reference them -- that was the reason for:
```
        for k, v in kwargs.items():
            setattr(self, k, v)
```

Reply:
I think dynamically adding unknown kwargs somewhat deviates from the whole purpose of classes no? I think allowing `setattr` creates situations where the caller has no idea what's in the object. What benefit is provided if the assignment happens? I think there's maybe a happy medium.

@donaldheppner and I had discussed this before so it may be good to get his opinion.

Reply:
😢  what happened?

Reply:
@cdkini to me, the benefit from before was that the configurations vary, but the top-level DictDot allows both dot and [] access.  We get this for known configuration parameters, so we also want it for unknown ones (from
```
    class Meta:
        unknown = INCLUDE
```
pattern).  For as long as we allow kwargs, it seems to be very nice to know that all fields have dot and dict.  I also feel that for consistency we should keep things same, and then change altogether; otherwise, when we get to the point of considering a change, we will not know why the current implementation is inconsistent and lose time wondering and trying to recall what happened.

Reply:
@donaldheppner I think we're very close to getting back the dataclass implementation - I'd request that we adhere to the current pattern for now and work to getting back this dataclass functionality down the line.

As this implementation (and the schema before it) blocks the rest of the sprint, I think this implementation works for now. If you feel strongly about the dataclasses, let's talk - it might extend the points of the ticket but it's perhaps worth looking into.

Reply:
@alexsherstinsky I'm a bit torn here. This won't be merged until tomorrow so I shall defer until then. I see the merits in your approach and don't want to go against the grain but I really dislike the dynamic nature of **kwargs.

Let's maybe quickly catch up tomorrow to lock this down.

Reply:
Adding the ability for others to add arbitrary attributes to top-level config objects kills our ability to evolve the platform -- when you add a new field you have no idea if it will conflict with a field that is already there because you have no idea what fields someone else has defined!

Why are we allowing arbitrary kwargs?

Reply:
Resolving this as we have decided to go with the `setattr` approach for now and push off larger scale refactors for later.